Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -190,10 +190,16 @@ def ncurses_define
 
   # @private
   def userpaths!
-    paths = ORIGINAL_PATHS.map { |p| p.realpath.to_s rescue nil } - %w[/usr/X11/bin /opt/X11/bin]
-    self["PATH"] = paths.unshift(*self["PATH"].split(File::PATH_SEPARATOR)).uniq.join(File::PATH_SEPARATOR)
+    paths = self["PATH"].split(File::PATH_SEPARATOR)
+    # put Superenv.bin and opt path at the first
+    new_paths = paths.select { |p| p.start_with?("#{HOMEBREW_REPOSITORY}/Library/ENV") || p.start_with?("#{HOMEBREW_PREFIX}/opt") }
     # XXX hot fix to prefer brewed stuff (e.g. python) over /usr/bin.
-    prepend_path "PATH", HOMEBREW_PREFIX/"bin"
+    new_paths << "#{HOMEBREW_PREFIX}/bin"
```

Comment:
Rethink a bit. I think this is required. For example, `depends_on :python` will collect `/usr/local/bin/` instead of `/usr/local/opt/python/bin` while here I only ensure opt path coming at first. So by adding `/usr/local/bin`, the brew python will be chosen over system python.